### PR TITLE
[dist] refactoring of obsapisetup

### PR DESCRIPTION
**Pull request by M0ses**
* splitted functional part of init script obsapisetup to have one base
  for install script in appliance and when installing package obs-api on
  a preinstalled system
* setup-appliance.sh also usable standalone
* shortname is used as subjectAltName in SSL cert
* improved handling of cert as CA (update-ca-certificates) 
* added disable_api_tests macro for faster building
* SEE https://build.opensuse.org/project/show/home:M0ses:branches:OBS:Server:Unstable for full appliance

### CODE REVIEW

**Comment by adrianschroeter** [comment]
Just had a brief look, not really reviewable for me. 
But I trust it given that you have tested it in your OBS project. So LGTM.

File: bootstrap.sh [hunk 1]
```diff
@@ -20,45 +60,67 @@ gem install bundler
...
+for dir in /srv/obs /srv/www/obs /srv/www/obs/overview/ /usr/lib/obs /srv/www/obs/api/log/ /srv/www/obs/api/tmp;do
+  [ -d $dir ] || mkdir -p $dir
+done 
+
+for srv in apache2 mysql memcached;do
+  systemctl enable $srv.service
+done
+
+echo -e "\nInstalling api components\n"
+cp -a /vagrant/src/api /srv/www/obs
```
Comment (hennevogel) [inline bootstrap.sh]: How are my changes to the outside code are ever going to reach the running app?

**Comment by M0ses** [comment]
I was not taking care about this problem. I´m not aware of your  workflow in frontend development

The second commit was accidently added to the PR after pushing to my own forked repo. Sorry.

 Anyway I would like to discuss this commit, because I would like to have a  possibility to verfy changes on init-scripts or in deployment very quickly and this seemed to be a good way for me, but it breaks frontend workflow - Is there a possiblity to use multiple Vagrantfile or possibility for multiple provisioners ?

If not, we could use symlinks to have multiple "profiles"

**Comment by hennevogel** [comment]
The current Vagrant box is a development environment and in no way related to the appliances we ship. If you are looking into using Vagrant to do something else I would suggest you look up the [multi-machine](https://docs.vagrantup.com/v2/multi-machine/index.html) feature of Vagrant.

**Comment by M0ses** [comment]
great, thanks a lot  I will prepare 2 new PRs

**Pull request closed without merging**